source/_posts/Docker-Swarm.md

```diff
@@ -0,0 +1,346 @@
+```
+Je kunt de schaling ook regelen in de UI van Portainer.
+<img src="/images/Docker-Swarm-scaling.png" />
+
+Mocht je nu toch te weinig capaciteit hebben in je cluster kun je eenvoudig een nieuwe virtuele machine inrichten met Ubuntu en Docker erop installeren. Hierna voer je het Docker Swarm join commando uit op de server en deze zal het bestaande cluster uitbreiden met de extra capaciteit. 
+```
+docker swarm join \
+    --token SWMTKN-1-3pu6hszjas19xyp7ghgosyx9k8atbfcr8p2is99znpy26u2lkl-1awxwuwd3z9j1z3puu7rcgdbx \
+    172.17.0.2:2377
+```
+Mocht je Docker Swarm op Azure hebben uitgerold kun je gebruik maken van een vm scale sets om automatisch nodes bij je cluster te zetten. Ik heb hier verder nog geen ervaring mee maar ga hier zeker mee aan de slag.
+
+Omdat ik Docker Swarm gebruik kan ik de volgende punten afvinken van mijn lijstje:
+- De oplossing moet schaalbaar zijn en bestand zijn tegen het uitvallen of offline gaan van servers. 
+    -- In het cluster zitten 3 managers dus er kan er 1 offline gaan volgens de berekening: "3 managers - 1 = 2  2/2 = 1. 
+    -- Ook hebben we 2 workers welke de containers kunnen hosten. Mocht er 1 offline gaan worden de containers opnieuw gestart op de andere node.
+    -- De manager heeft onder andere als taak om er voor te zorgen dat de containers draaien op 1 of meerdere nodes.
+- Ik wil de oplossing kunnen hosten bij een Cloud provider, Hosting provider of lokaal om te kunnen testen.
+    -- Omdat Docker Swarm standaard in elke Docker installatie zit kan het zonder verdere installatie van tools gebruikt worden overal waar je Docker hebt geïnstalleerd.  
+
+Wat is Let's Encrypt
+---
+Let's Encrypt is een certificaatautoriteit opgericht op 16 april 2016. Het geeft X.509 certificaten uit voor het Transport Layer Security (TLS) encryptie-protocol, zonder dat dit kosten met zich meebrengt. De certificaten worden uitgegeven via een geautomatiseerd proces dat is ontworpen om het tot nu toe complexe proces van handmatige validatie, ondertekening, installatie en hernieuwing van certificaten voor beveiligde websites te elimineren. (Wikipedia)
+
+Voor meer informatie over Let's Encrypt zie https://letsencrypt.org/
+
+Wat is Traefik
+---
+Traefik is een opensource router welke speciaal is ontworpen voor container oplossingen. Traefik wordt als global service op elke manager gedeployed op het cluster. Dit wil zeggen elke node met als rol manager krijgt een Traefik container. De reden dat Traefik op de manager nodes gedeployed dient te worden is dat de Docker api wordt uitgelezen. Zodra er een container bij komt en deze is geconfigureerd met de Traefik labels kan Traefik de labels van de container uitlezen en een virtuele host aanmaken voor de container en een SSL-certificaat aanvragen bij Let's Encrypt. Zodoende is de container beschikbaar voor de buitenwereld met een SSL-certificaat.
+
+Zie hier een voorbeeld Docker-compose file om een Traefik container te deployen als stack\*\* op het Docker Swarm cluster.
```

Comment:
Wat kan ik met een Docker compose file?

Reply:
3.Docker-Compose is een hulpmiddel voor het definiëren en uitvoeren van Docker-toepassingen met meerdere containers